clients/src/main/java/org/apache/kafka/clients/consumer/ConsumerConfig.java

```diff
@@ -727,6 +731,20 @@ private void checkGroupRemoteAssignor() {
         }
     }
 
+    @SuppressWarnings("unchecked")
+    private void checkPartitionAssigmentStrategy() {
+        List<Object> assignmentStrategies = (List<Object>) get(PARTITION_ASSIGNMENT_STRATEGY_CONFIG);
+        if (getString(GROUP_PROTOCOL_CONFIG).equalsIgnoreCase(GroupProtocol.CONSUMER.name())) {
```

Comment:
I wonder if it would help readability to encapsulate this in something along the lines of: 
```suggestion
        if (isConsumerProtocol()) {
```
And then a similar `isClassicProtocol()`. The trick is that we already have protocol checks in 2 places (this and `checkGroupRemoteAssignor`), and I expect we will need it again as we keep adding other consistency validations.

Reply:
It's a good idea, I will extract these method.